python/cudf/cudf/core/index.py

```diff
@@ -2263,15 +2267,25 @@ def __init__(
                 "dtype parameter is supported"
             )
 
+        valid_dtypes = tuple(
+            f"timedelta64[{res}]" for res in ("s", "ms", "us", "ns")
+        )
+        if dtype not in valid_dtypes:
+            raise TypeError("Invalid dtype")
```

Comment:
same comment as above for having this be in `as_column`

Reply:
Addressed similar to above.